pkg/vsphere/guestinfo/encode.go

```diff
@@ -0,0 +1,282 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package guestinfo
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+const (
+	// DefaultTagName value
+	DefaultTagName = "vic"
+	// DefaultPrefix value
+	DefaultPrefix = ""
+)
+
+const (
+	// Invalid value
+	Invalid = 1 << iota
+	// Hidden value
+	Hidden
+	// ReadOnly value
+	ReadOnly
+	// ReadWrite value
+	ReadWrite
+	// NonPersistent value
+	NonPersistent
```

Comment:
NonPersistent is because it's the rare case where Persistent is the general case. If we use nonpersistent as the bool it can be omitted from the tag set for most fields. This is for conciseness and readability.

The reason I don't suggest the same thing for read-only/read-write is that people tend to assume that a value is read-write, yet that's not our common case. Being explicit as to which for that pairing should improve readability.